### PR TITLE
Pcc02

### PR DESCRIPTION
**Pull request by jatinsinghal1921**
ATTENTION: before clicking "Create Pull Request" please submit some meta data, thanks!

Difficulty level (1-10): []
Estimated time spent (hours): []
Completed (yes/no): []
I stretched my coding skills (if yes what did you learn?): []
Other feedback (what can we improve?): []

### CODE REVIEW

**Comment by jatinsinghal1921** [comment]
Difficulty level (1-10): [1]
Estimated time spent (hours): [1]
Completed (yes/no): [Yes]
I stretched my coding skills (if yes what did you learn?): []
Other feedback (what can we improve?): []

File: 02/jatinsinghal1921/game.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+# Random Selection of Letters from the Pouch [List of repetitive Alphabets]
+def draw_letters():
+    return random.sample(POUCH, NUM_LETTERS)
+
+
+# Storing the picked letters in dictionary format. 
+def letters_picked_in_dict_form(letters_picked):
+    letters_picked_dict = {}
+    for letter in letters_picked:
+        if letter in letters_picked_dict:
```
Comment (bbelderbos) [inline 02/jatinsinghal1921/game.py]: you can avoid this if/else using `defaultdict` from the `collections` module

File: 02/jatinsinghal1921/game.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            letters_picked_dict[letter] = letters_picked_dict[letter] + 1
+        else:
+            letters_picked_dict[letter] = 1
+
+    return letters_picked_dict
+
+
+# Selecting the words in dictionary which can be formed from the picked letters
+def possible_words_in_dictionary(letters_picked_dict):
```
Comment (bbelderbos) [inline 02/jatinsinghal1921/game.py]: consider using `itertools.permutations`

File: 02/jatinsinghal1921/game.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+
+# re-use from challenge 01
+def calc_word_value(word):
+    """Calc a given word value based on Scrabble LETTER_SCORES mapping"""
+    return sum(LETTER_SCORES.get(char.upper(), 0) for char in word)
+
+
+# re-use from challenge 01
+def max_word_value(words_list):
```
Comment (bbelderbos) [inline 02/jatinsinghal1921/game.py]: you could use `max(words, key=calc_word_value)`

**Comment by bbelderbos** [comment]
Good work, few suggestions. You can compare solutions on our community branch

**Pull request merged**